core/src/main/java/io/camunda/migrator/HistoryMigrator.java

```diff
@@ -146,23 +159,28 @@ private void migrateDecisionDefinitions() {
     LOGGER.info("Migrating decision definitions");
 
     DecisionDefinitionQuery legacyDecisionDefinitionQuery = repositoryService.createDecisionDefinitionQuery()
+        .orderByDeploymentTime()
+        .asc()
         .orderByDecisionDefinitionId()
         .asc();
 
-    String latestLegacyId = idKeyMapper.findLatestIdByType(IdKeyMapper.TYPE.HISTORY_DECISION_INSTANCE);
-    if (latestLegacyId != null) {
-      LOGGER.debug("Resume functionality not available for decision definitions - starting from beginning");
+    Date deployedAfter = getLatestStartDateByType(HISTORY_DECISION_DEFINITION);
+    if (deployedAfter != null) {
+      legacyDecisionDefinitionQuery.deployedAfter(deployedAfter);
     }
 
-    new Pagination<DecisionDefinition>()
-        .pageSize(getPageSize())
+    new Pagination<DecisionDefinition>().pageSize(getPageSize())
         .query(legacyDecisionDefinitionQuery)
         .callback(legacyDecisionDefinition -> {
           String legacyId = legacyDecisionDefinition.getId();
           LOGGER.debug("Migrating legacy decision definition with id: [{}]", legacyId);
           DecisionDefinitionDbModel dbModel = decisionDefinitionConverter.apply(legacyDecisionDefinition);
           decisionDefinitionMapper.insert(dbModel);
-          insertKeyIdMapping(legacyId, dbModel.decisionDefinitionKey(), IdKeyMapper.TYPE.HISTORY_DECISION_INSTANCE);
+          Date deploymentTime = repositoryService.createDeploymentQuery()
```

Comment:
This newly introduced C7 query should go into `C7Client` as well.

Reply:
Was thinking of doing the rest of these with https://github.com/camunda/camunda-bpm-platform/issues/5308 , but will move this one now and clean up the rest with the follow up task